frontend/src/pages/Profile.jsx

```diff
@@ -61,26 +69,48 @@ class Profile extends Component {
         )
     }
 
+    componentDidMount() {
+        this.getProfile();
+    }
+
+    getProfile() {
+        var urlPath = "/api/author/"
+        var authorId = store.getState().loginReducers.userId.split("thor/");
+        const path = urlPath + authorId[1], requireAuth = true;
+        HTTPFetchUtil.getRequest(path, requireAuth)
+        .then((httpResponse) => {
+            if (httpResponse.status === 200) {
+                httpResponse.json().then((results) => {
+                    this.setState( {
+                        profiledata: results
+                    })
+                })
+            }
+        });
+    }
+
 	render() {
-	return(
-		    <Container>
+	    return(
+            <Container>
                 <SideBar/>
-                    <div className="profile">
-                        <ProfileBubble
-                        profileBubbleClassAttributes={"ui centered top aligned circular bordered small image"}/>
-                        <br/><div className="profile-username">htruong1</div>
-                            <button class="positive ui button">
-                            <i class="user plus icon"></i>
-                            Request Friend
-                            </button>
-                    
-                    <div>
-                        <Tab panes={this.showPanes()}></Tab>
-                    </div>
-                    </div>
+                <div className="profile">
+                    <br/>
+                    <ProfileBubble
+                    profileBubbleClassAttributes={"ui centered top aligned circular bordered small image"}/>
+                    <br/><div className="profile-username">{this.state.profiledata.displayName}</div>
+                <div>
+                    <Tab panes={this.showPanes()}></Tab>
+                </div>
+                </div>
             </Container>
 	    )
     }
 }
 
-export default Profile;
+const mapStateToProps = (state) => {
+    return {
```

Comment:
I don't think I see this props value (this.props.userId from mapStateToProps) being used anywhere. Are you putting this here in the case you need it for the future?

Reply:
left it in b/c I thought it was necessary for grabbing user id from store, removed